doc/index.rst

```diff
@@ -47,14 +45,15 @@ User manual
     auto_examples/index
     developing.rst
 
-Module reference
------------------
-
 .. currentmodule:: joblib
 
+Module reference
```

Comment:
I need to duplicate the title here and below (not really sure why). It could be because we are within an `autosummary` and the caption is not used then.